zaza/openstack/charm_tests/ceph/rbd_mirror/tests.py

```diff
@@ -124,6 +136,119 @@ def wait_for_mirror_state(self, state, application_name=None,
                 # all images with state has expected state
                 return True
 
+    def get_cinder_rbd_mirroring_mode(self,
+                                      cinder_ceph_app_name='cinder-ceph'):
+        """Get the RBD mirroring mode for the Cinder Ceph pool.
+
+        :returns: A string representing the RBD mirroring mode. It can be
+                  either 'pool' or 'image'.
+        """
+        DEFAULT_RBD_MIRRORING_MODE = 'pool'
+
+        rbd_mirroring_mode_config = zaza.model.get_application_config(
+            cinder_ceph_app_name).get('rbd-mirroring-mode')
+        if rbd_mirroring_mode_config:
+            rbd_mirroring_mode = rbd_mirroring_mode_config.get(
+                'value', DEFAULT_RBD_MIRRORING_MODE).lower()
+        else:
+            rbd_mirroring_mode = DEFAULT_RBD_MIRRORING_MODE
+
+        return rbd_mirroring_mode
+
+    def create_cinder_volume(self, session, from_image=False):
+        """Create Cinder Volume from image.
+
+        :rtype: :class:`Volume`.
+        """
+        def get_glance_image(session):
+            glance = openstack.get_glance_session_client(session)
+            images = openstack.get_images_by_name(glance, CIRROS_IMAGE_NAME)
+            if images:
+                return images[0]
+            logging.info("Failed to find {} image, falling back to {}".format(
+                CIRROS_IMAGE_NAME,
+                LTS_IMAGE_NAME))
+            return openstack.get_images_by_name(glance, LTS_IMAGE_NAME)[0]
+
+        def create_volume_type(cinder):
+            try:
+                vol_type = cinder.volume_types.find(name='repl')
+            except cinder_exceptions.NotFound:
+                vol_type = cinder.volume_types.create('repl')
+                vol_type.set_keys(metadata={
+                    'volume_backend_name': 'cinder-ceph',
```

Comment:
Shouldn't `'cinder-ceph'` actually come from an argument instead of being hard-coded, like for `get_cinder_rbd_mirroring_mode()`?

Reply:
Yes, you're right. I'll change it.